.github/workflows/check-code-quality.yml

```
@@ -75,3 +75,16 @@ jobs:
             git status --porcelain --untracked-files=no
             exit 1
           fi
+      
+      - name: Check generated code
+        if: success()
+        run: |
+          pip install -r code_generation/requirements.txt
+          python code_generation/code_gen.py
+          if [ -n "$(git status --porcelain)" ]; then
+            echo "The following files are outdated or were manually updated:"
+            git status --porcelain
+            exit 1
+          else
+            echo "All the generated files are up to date."
+          fi
```

newline at EOF (good practice due to not all editors handling this nicely)

```suggestion
          fi

```

---

Got it.